spec/models/course/discussion/topic_spec.rb

```diff
@@ -155,5 +155,42 @@
         it { is_expected.to contain_exactly(annotation, comment) }
       end
     end
+
+    describe '.destroy' do
+      it 'destroys successfully' do
+        expect(topic.destroy).to be_truthy
+      end
+
+      context 'when it has multiple posts' do
+        let!(:post) { create(:course_discussion_post, topic: topic.discussion_topic) }
+        let!(:children_posts) do
+          create_list(:course_discussion_post, 2, parent: post, topic: topic.discussion_topic)
+        end
+
+        it 'destroys successfully with posts' do
+          expect(topic.reload.destroy).to be_truthy
+          all_post_ids = [post, *children_posts].map(&:id)
+          expect(Course::Discussion::Post.where(id: all_post_ids).exists?).to be_falsey
+        end
+
+        context 'when the posts have votes' do
+          before(:each) do
+            post.cast_vote!(user, 1)
```

Comment:
Instead of explicitly calling `#cast_vote!` here, we can add traits to the `posts` factory (eg. `post_count`).